src/promptflow-devkit/promptflow/_sdk/_utilities/tracing_utils.py

```diff
@@ -330,3 +334,41 @@ def aggregate_trace_count(all_spans: typing.List[Span]) -> typing.Dict[TraceCoun
             trace_count_summary[key] = trace_count_summary.get(key, 0) + 1
 
     return trace_count_summary
+
+
+class TraceTelemetryHelper:
+    """Helper class for trace telemetry in prompt flow service."""
+
+    LOG_INTERVAL_SECONDS = 60
+    TELEMETRY_ACTIVITY_NAME = "pf.telemetry.trace_count"
+    CUSTOM_DIMENSIONS_TRACE_COUNT = "trace_count"
+
+    def __init__(self):
+        self._telemetry_logger = get_telemetry_logger()
+        self._lock = multiprocessing.Lock()
+        self._summary: typing.Dict[TraceCountKey, int] = dict()
+        self._thread = threading.Thread(target=self._schedule_flush, daemon=True)
+        self._thread.start()
+
+    def _schedule_flush(self) -> None:
+        while True:
+            time.sleep(self.LOG_INTERVAL_SECONDS)
+            self.log_telemetry()
+
+    def append(self, summary: typing.Dict[TraceCountKey, int]) -> None:
+        with self._lock:
+            for key, count in summary.items():
+                self._summary[key] = self._summary.get(key, 0) + count
+
+    def log_telemetry(self) -> None:
+        with self._lock:
```

Comment:
may we can only lock the process to cleanup the _summary, and not include _telemetry_logger.info() in the lock period

Reply:
good catch! updated to not lock log telemetry operation.